src/ReactiveDomain.Testing/Specifications/MockRepositorySpecification.cs

```diff
@@ -27,13 +33,75 @@ public MockRepositorySpecification()
             ConfiguredConnection = new ConfiguredConnection(StreamStoreConnection, StreamNameBuilder, EventSerializer);
 
             var connectorBus = new InMemoryBus("connector");
-            StreamStoreConnection.SubscribeToAll(evt => {
-                if(evt is ProjectedEvent) { return;}
-                connectorBus.Publish((IMessage) EventSerializer.Deserialize(evt));
+            StreamStoreConnection.SubscribeToAll(evt =>
+            {
+                if (evt is ProjectedEvent) { return; }
+                connectorBus.Publish((IMessage)EventSerializer.Deserialize(evt));
+            });
+            RepositoryEvents = new TestQueue(connectorBus, new[] { typeof(Event) });
+        }
+
+        public MockRepositorySpecification(string schema)
+        {
+            _schema = schema;
+            StreamNameBuilder = new PrefixedCamelCaseStreamNameBuilder(schema);
+            StreamStoreConnection = new MockStreamStoreConnection(Schema);
+            StreamStoreConnection.Connect();
+            EventSerializer = new JsonMessageSerializer();
+            MockRepository = new StreamStoreRepository(StreamNameBuilder, StreamStoreConnection, EventSerializer);
+
+            ConfiguredConnection = new ConfiguredConnection(StreamStoreConnection, StreamNameBuilder, EventSerializer);
+
+            var connectorBus = new InMemoryBus("connector");
+            StreamStoreConnection.SubscribeToAll(evt =>
+            {
+                if (evt is ProjectedEvent) { return; }
+                connectorBus.Publish((IMessage)EventSerializer.Deserialize(evt));
+            });
+            RepositoryEvents = new TestQueue(connectorBus, new[] { typeof(Event) });
+        }
+        public MockRepositorySpecification(string schema, IStreamStoreConnection dataStore)
```

Comment:
Let's make this one the "main" constructor and have the other call it using `this`.